content/src/service/validations/Validations.ts

```diff
@@ -145,6 +145,21 @@ export class Validations {
           errors.push(`This hash was uploaded but is not referenced in the entity: ${unreferencedHash}`)
         )
 
+      if (entity.type === EntityType.PROFILE) {
+        // Validate all content files correspond to at least one avatar snapshot
+        if (Profile.validate(entity.metadata)) {
+          Array.from(entity.content.keys())
+            .filter((fileName) => {
+              const fileNameWithoutExtension = fileName.replace(/.[^/.]+$/, '')
+              const testIsSnapshot = (avatar: Avatar) => avatar.avatar.snapshots[fileNameWithoutExtension] !== undefined
+              return entity.metadata.avatars.find(testIsSnapshot) === undefined
+            })
+            .forEach((fileName) => {
+              errors.push(`This file is not expected: ${fileName}. Please, include only snapshot files.`)
+            })
```

Comment:
What do you think of using plain fors instead of chained callbacks? In big chunks, performance is best if we are not calling functions and also readability is better

```suggestion
          for (const fileName of entity.content.keys()) {
              const fileNameWithoutExtension = fileName.replace(/.[^/.]+$/, '')
              const testIsSnapshot = (avatar: Avatar) => avatar.avatar.snapshots[fileNameWithoutExtension] !== undefined
              if (entity.metadata.avatars.find(testIsSnapshot) === undefined) {
                errors.push(`This file is not expected: ${fileName}. Please, include only snapshot files.`)
              }
          }
```

Reply:
I agree readability is better and performance may improve too.

In this case I tried to do the same way as it was above (same method, multiple validations being done using these `filter` and `forEach` calls), otherwise it would end up with mixed approaches. 

If you think it worth to make these validations in a more clear way, I'm happy to open another PR for that.